crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -1140,9 +1145,10 @@ pub mod tests {
                         }))),
                 )
                 .await;
-        (mock_ecrow_subgraph_server, _mock_ecrow_subgraph)
+        mock_escrow_subgraph_server
     }
 
+    #[allow(clippy::too_many_arguments)]
     async fn create_sender_account(
```

Comment:
You could add #[bon] here, seems a good place since you hit too many arguments.

Reply:
I know this is gonna change in a lot of places, but you are already changing them to add None, so it's nice if we could simplify those calls.